### PR TITLE
[Connector API] Fix serialisation of script params in connector index service

### PR DESCRIPTION
**Pull request by jedrazb**
### Changes
- Represent Script `params` as a generic Map instead of writeable instance of `ConnectorConfiguration` as it seems to be causing serialisation issues in the transport
- Add `toMap` method to `ConnectorConfiguration` to represent the instance as generic map

This should fix following error:

We want to use a scripted update (to completely override a nested field) within ES index service for connectors. Sometimes the api request succeeds but sometimes it fails with:

```
[instance-0000000001] failed to serialize outbound message [Request{indices:data/write/update[s]}{38653}{false}{false}{false}] java.lang.IllegalArgumentException: can not write type [class org.elasticsearch.xpack.application.connector.ConnectorConfiguration] at org.elasticsearch.common.io.stream.StreamOutput.writeGenericValue(StreamOutput.java:843) ~[elasticsearch-8.13.0.jar:?] at org.elasticsearch.common.io.stream.StreamOutput.writeMap(StreamOutput.java:604) ~[elasticsearch-8.13.0.jar:?] at org.elasticsearch.common.io.stream.StreamOutput.lambda$static$10(StreamOutput.java:678) ~[elasticsearch-8.13.0.jar:?] at org.elasticsearch.common.io.stream.StreamOutput.writeGenericValue(StreamOutput.java:841) ~[elasticsearch-8.13.0.jar:?] at org.elasticsearch.common.io.stream.StreamOutput.writeMapWithConsistentOrder(StreamOutput.java:563) ~[elasticsearch-8.13.0.jar:?] at org.elasticsearch.script.Script.writeTo(Script.java:572)
```

We are using scripted update internally as we want to be able to completely override a configuration field within a Connector which is a `Map<String, ConnectorConfiguration>` represented as an object in the index mapping.

#### Investigation 
- We pass `configuration` param to the `UpdateRequest` script with `Map<String, ConnectorConfiguration>` and it's sometimes causing issue: `can not write type [class org.elasticsearch.xpack.application.connector.ConnectorConfiguration]`
- The stack trace references `Script.writeTo` that calls `writeMapWithConsistentOrder(params)` that calls `writeGenericValue` and we fail [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java#L843)
- Even though `ConnectorConfiguration` extends `Writeable`, the `writeGenericValue` won't be able to look up the class writer (defined in [WRITERS](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java#L636)) since `ConnectorConfiguration` is not a `GenericNamedWriteable` (and I'm not sure it should be)
- Therefore, let's use a generic map representation for nested parameter for Script

### CODE REVIEW

**Review by carlosdelest — APPROVED** [review]
LGTM

**Review by kderusso — APPROVED** [review]
I suspect we might be able to do something with the xcontent serialization here, but this seems like a fine workaround to your issue

File: docs/changelog/106060.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 106060
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply docs/changelog/106060.yaml:1]: @kderusso that bug is a regression that hasn't been deployed yet. For such cases we still need to use the `>bug` label right? If I stumble across similar scenario in the future, should I just delete the changelog generated by the automation?

[the remark this answers is not in the document]
Comment (kderusso) [reply docs/changelog/106060.yaml:1]: I would tag it as `>non-issue` - it's not a bug impacting customers so it doesn't need to be tagged in release notes.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/configuration/ConfigurationSelectOption.java [hunk 2]
```diff
@@ -60,6 +62,13 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public Map<String, Object> toMap() {
+        Map<String, Object> map = new HashMap<>();
+        map.put(LABEL_FIELD.getPreferredName(), label);
+        map.put(VALUE_FIELD.getPreferredName(), value);
+        return map;
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/configuration/ConfigurationSelectOption.java:69]: Getting back to addressing the feedback in a followup PR, I realised I can't use Map.of here as value can be `null`

**Pull request merged**